kategory/src/main/kotlin/kategory/data/Option.kt

```diff
@@ -43,14 +43,15 @@ sealed class Option<out A> : OptionKind<A> {
 
     /**
      * Returns true if the option is [None], false otherwise.
-     * Used only for performance instead of fold.
+     * @note Used only for performance instead of fold.
      */
     internal abstract val isEmpty: Boolean
 
     /**
      * Returns true if the option is an instance of $some, false otherwise.
+     * @note Used only for performance instead of fold.
      */
-    val isDefined: Boolean = !isEmpty
+    internal val isDefined: Boolean = !isEmpty
```

Comment:
if these are internal they can't be used elsewhere. Do we want this APIs to actually be here for any reason?
Also I realized writing the docs for `Option` that we don't need subtypes overrides at all because:
```kotlin
val isDefined: Boolean = this is Some
val isEmpty: Boolean = this is None
```

Reply:
Microperf, as checking the boolean is chaper than checking the types